### ISSUE TITLE
policy: keep parser type of first rule that applies at L7 even if all traffic is allowed at L7

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#3697]
Consider the following rule: 

```
specs:
- endpointSelector:
    matchLabels:
      id: app1
  ingress:
  -  toPorts:
    - ports:
      - port: "9093"
        protocol: TCP
      rules:
        http:
        - method: ^GET$
          path: ^/metrics$
  - toPorts:
    - ports:
      - port: "9093"
        protocol: TCP
```

While this rule means that we allow all at L7, because there is L7 policy for that port, we must redirect through the proxy nonetheless. Do not set the proxy type in the L4Filter for this port to 'None'; instead, retain the proxy type of the first L7 rule for this port.

**Comment by ianvernon** [comment]
I disagree that this needs an e2e test, this is entirely unit testable.

**Comment by ianvernon** [comment]
I'm unsure about the semantics of `L7RulesPerEp` and how it interacts with `L7Parser` in the `L4Filter`. Need to sit down and discuss this to progress further.

**Comment by tgraf** [comment]
> I'm unsure about the semantics of L7RulesPerEp and how it interacts with L7Parser in the L4Filter. Need to sit down and discuss this to progress further.

Will review in details and provide feedback.

**Comment by ianvernon** [comment]
Fixed by #3732.
